src/components/contactInfoForm/ContactInfoForm.tsx

```diff
@@ -1,25 +1,86 @@
-import React from "react";
-import { useForm } from "react-hook-form";
+import React, { useState, useEffect } from "react";
+import { FieldValues, useForm } from "react-hook-form";
 import { yupResolver } from "@hookform/resolvers/yup";
 import i18next from "localization/en/en.json";
+import { toast } from "react-toastify";
+import { useSelector } from "react-redux";
 import LayoutElementWithTitle from "components/layoutElementWithTitle/LayoutElementWithTitle";
 import propsDataCollection from "components/contactInfoForm/staticData";
 import FormSubmitButton from "components/UI/buttons/formSubmitButton/FormSubmitButton";
 import { SButtonWrapper } from "components/profileEditForm/styles";
 import contactEditFormValidationSchema from "validation/contactEditFormValidationSchema";
+import { IContactInfoState } from "components/contactInfoForm/typesDef";
+import {
+  useGetUserQuery,
+  useUpdateUserMutation,
+} from "services/contactInfo/contactInfoAPI";
+import { RootState } from "store/store";
 
 function ContactInfoForm() {
+  const [initialState, setInitialState] = useState<IContactInfoState>({
+    email: "",
+    firstName: "",
+    lastName: "",
+    phone: null,
+  });
+
+  const userId = useSelector<RootState>(({ user }) => user.id);
+  const {
+    data,
+    isSuccess,
+    isError: getUserError,
+  } = useGetUserQuery(Number(userId));
+  const [
+    updateUser,
+    { error, isError: submitError, isSuccess: submitSuccess },
+  ] = useUpdateUserMutation();
+
+  useEffect(() => {
+    if (isSuccess)
+      setInitialState({
+        email: data.email,
+        firstName: data.firstName,
+        lastName: data.lastName,
+        phone: data.phone === null ? null : data.phone.slice(1),
+      });
+  }, [data, isSuccess]);
+
   const {
     handleSubmit,
+    reset,
     control,
     formState: { errors },
-  } = useForm({
+  } = useForm<FieldValues>({
     resolver: yupResolver(contactEditFormValidationSchema),
+    defaultValues: initialState,
   });
 
-  const onSubmit = (data: any) => {
-    // eslint-disable-next-line no-console
-    console.log(data);
+  useEffect(() => {
+    reset(initialState);
+  }, [reset, initialState]);
+
+  useEffect(() => {
+    if (submitSuccess) toast.success("changes saved");
+    //  @ts-ignore
+    if (submitError) toast.error(error.data.message[0]);
+    if (getUserError) toast.error("Sorry, user not found");
+    // eslint-disable-next-line
+  }, [submitSuccess, submitError, getUserError]);
+
+  const onSubmit = (formData: any) => {
+    const phone =
+      formData.phone === null || formData.phone === ""
```

Comment:
formData.phone && `+${formData.phone}`; I think it should work

Reply:
it's not work because in case if it comes empty string this expression will return empty string, but I need null.
left unchanged

Reply:
formData.phone ?? +${formData.phone} - no? it should be something

Reply:
Is `const phone = !formData.phone ? null : `+${formData.phone}`;` will be ok?